crates/matrix-sdk-crypto/src/store/memorystore.rs

```diff
@@ -56,7 +56,7 @@ pub struct MemoryStore {
     inbound_group_sessions: GroupSessionStore,
     outbound_group_sessions: StdRwLock<BTreeMap<OwnedRoomId, OutboundGroupSession>>,
     private_identity: StdRwLock<Option<PrivateCrossSigningIdentity>>,
-    tracked_users: StdRwLock<Vec<TrackedUser>>,
+    tracked_users: StdRwLock<HashMap<OwnedUserId, TrackedUser>>,
```

Comment:
This could just be a `HashSet`, right? It doesn't seem there's any lookup by user id using this `tracked_users` field :eyes:

Reply:
As discussed IRL, we want it to be unique by `UserId`, not the whole `TrackedUser` so we do need the `HashMap`